pkg/appvalidation/business_case.go

```diff
@@ -76,3 +76,143 @@ func BusinessCaseForUpdate(businessCase *models.BusinessCase) error {
 	}
 	return nil
 }
+
+// BusinessCaseForSubmit checks if it's a valid business case to update
+func BusinessCaseForSubmit(businessCase *models.BusinessCase, existingBusinessCase *models.BusinessCase) error {
+	// We return an empty id in this error because the business case hasn't been created
+	expectedErr := apperrors.NewValidationError(
+		errors.New("business case failed validations"),
+		businessCase,
+		businessCase.ID.String(),
+	)
+
+	if businessCase.Status == models.BusinessCaseStatusSUBMITTED {
+		if existingBusinessCase.Status != models.BusinessCaseStatusDRAFT &&
+			existingBusinessCase.Status != models.BusinessCaseStatusSUBMITTED {
+			expectedErr.WithValidation("Status", "Cannot be SUBMITTED")
```

Comment:
The line above seems to indicate that a submitted business case is allowable, but this error says it is not.

Reply:
This validation should only be triggered when it is past `SUBMITTED`, and thus not submit-able. Unless I'm missing something a case?

Reply:
lmao I was thinking "SUBMITTED", the constant, not submitted, the verb.